### ISSUE TITLE
[DX] Offer to backup and delete existing config to proceed with installation

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#6062]
I work with a lot of local instances, and when I wipe the db to start over, I always get this screen:

![image](https://user-images.githubusercontent.com/2423362/232227698-26226264-17be-471d-b6a3-f78f57229d30.png)

I would like us to add an option in this screen to allow backing up the existing configuration files, and then deleting them, so that people don't have to do it manually.

There could be an "I understand the risk" checkbox and the submit button could be a red "danger" one, to communicate the irreversibility of things.

Another great suggestion by @docwilmot:

Could the update process offer options?:
> Your config folder is not empty (has existing files). Would you like to:
>- backup the existing config files and continue with a fresh install
>- import the existing config files 
>- delete the existing files and continue with a fresh install

**Comment by kiamlaluno** [comment]
I am not sure about backing the configuration files, but I would expect the installation code to be able to delete those files, since that is the directory where it then creates files.

Rather than backing up the files, would not be better to "migrate" the configuration files?

**Comment by kiamlaluno** [comment]
What I mean by "migrate" is that the installation ends as usual and then imports the values contained in those files.

**Comment by klonos** [comment]
>I am not sure about backing up the configuration files...

Well, that's the thing @kiamlaluno: we don't know what leads people to this page. In my case, I know what it was - I intentionally deleted the db in order to start over. However, people may get here because of other reasons, such as the db server being temporarily down or something like that. In that case, we don't want people rushing through clicking buttons and ending up wiping their config irreversibly. That's why I am suggesting to backup the current config files - in case people need to revert things.

> What I mean by "migrate" is that the installation ends as usual and then imports the values contained in those files.

That does sound like a thing, however there may be config there that relates to contrib modules, so until we have #5163, that may not work as expected.

**Comment by kiamlaluno** [comment]
I hope users are not redirected to the installer simply because the database server was temporary down.
The database classes should be able to recognize when a database table does not exist or when the database server is unresponsive.

**Comment by klonos** [comment]
Let me test that and get back to you...

**Comment by klonos** [comment]
...tested this by setting up a fresh installation of Backdrop using lando, then stopped the lando instance with `lando stop`, then added `database: disabled` in the `services:` section of the .lando.yml file, then started the instance with `lando start`. Refreshed, and got this:

![image](https://user-images.githubusercontent.com/2423362/232248286-d6c6384d-bea5-4c06-b070-d2598acdc474.png)

So yeah, not the same thing if the db server goes away.

**Comment by stpaultim** [comment]
This would be very useful for me as a developer. Probably not something that most users need, but for those of us who need it, it would be REALLY nice and I don't think it's an option for contrib as it must be there on installation. 

It's not clear to me about the need to backup the files. I know that in my case, I always run into this when I'm starting over with a site so I am just going to delete them. Of course we need to think this through to be sure we are not making it too easy to make a big mistake.

**Comment by klonos** [comment]
> Of course we need to think this through to be sure we are not making it too easy to make a big mistake.

That's my thinking as well 👍🏼

**Comment by kiamlaluno** [comment]
Essentially, what happens with the current code is the following:

- The database is wiped out by the site administrator
- The next time the site administrator visits the site, the site redirects to the installer
- The installer notices the configuration directory is not empty and it shows a message

Given that, what is the purpose of creating a backup?  
Should the backup be done on request by the site administrator, or automatically?

> Of course we need to think this through to be sure we are not making it too easy to make a big mistake.

Since Backdrop shows the following screen, that should not be a possibility.

![screenshot](https://user-images.githubusercontent.com/29851/232416298-87f8edfc-0398-4096-a81c-ddf1348aa1a0.png)

Eventually, the showed message should also consider the fact the *settings.php* file could have been edited with the wrong database name and suggest to check that.  
The database could not exist because either it has been wiped out or because the database name is not correct. In the first case, the site administrator probably wants to start over; in the second case, it is just a mistake and the site administrator does not want to start over.

**Comment by kiamlaluno** [comment]
In the case the database has been wiped out by the site administrator, could the site administrator use the existing configuration files in some way? I guess that is only possible if the site administrator also has a backup of the database table.

**Comment by klonos** [comment]
> The database is wiped out by the site administrator

Yes, I think that this is key: Backdrop in this situation has all the access it needs to write/read to/from the database, but it finds it to be empty. Whereas in the instance of the PDO Exception it cannot connect to the db server at all. In that second instance, you are not being automatically redirected to the installer, and if you manually navigate to `core/install.php`, you are presented with a set of useful instructions.

> Eventually, the message shown in that screen should also consider the fact the settings.php file could have been edited with the wrong database name and suggest to check that.

Yes, good thinking there - that's definitely another possibility.

> ...could the site administrator use the existing configuration files in some way? I guess that is only possible if the site administrator also has a backup of the database table; in this case, the site could be restored to the state it had before the database was wiped out.

Exactly. That's why the safest thing to do would be to keep a backup of the config files - just in case.

**Comment by docwilmot** [comment]
Could the update process offer options?:
> Your config folder is not empty (has existing files). Would you like to:
---- backup the existing config files and continue with a fresh install
---- import the existing config files 
---- delete the existing files and continue with a fresh install
